packages/beacon-node/src/metrics/validatorMonitor.ts

```diff
@@ -1,16 +1,32 @@
-import {computeEpochAtSlot, AttesterStatus, parseAttesterFlags} from "@lodestar/state-transition";
-import {Logger} from "@lodestar/utils";
-import {allForks, altair} from "@lodestar/types";
-import {ChainForkConfig} from "@lodestar/config";
-import {MIN_ATTESTATION_INCLUSION_DELAY, SLOTS_PER_EPOCH} from "@lodestar/params";
+import {
+  computeEpochAtSlot,
+  AttesterStatus,
+  parseAttesterFlags,
+  CachedBeaconStateAllForks,
+  CachedBeaconStateAltair,
+  parseParticipationFlags,
+  computeStartSlotAtEpoch,
+  getBlockRootAtSlot,
+  ParticipationFlags,
+} from "@lodestar/state-transition";
+import {Logger, MapDef, MapDefMax, toHex} from "@lodestar/utils";
+import {RootHex, allForks, altair} from "@lodestar/types";
+import {ChainConfig, ChainForkConfig} from "@lodestar/config";
+import {ForkSeq, INTERVALS_PER_SLOT, MIN_ATTESTATION_INCLUSION_DELAY, SLOTS_PER_EPOCH} from "@lodestar/params";
 import {Epoch, Slot, ValidatorIndex} from "@lodestar/types";
 import {IndexedAttestation, SignedAggregateAndProof} from "@lodestar/types/phase0";
 import {LodestarMetrics} from "./metrics/lodestar.js";
 
 /** The validator monitor collects per-epoch data about each monitored validator.
  * Historical data will be kept around for `HISTORIC_EPOCHS` before it is pruned.
  */
-const HISTORIC_EPOCHS = 4;
+const MAX_CACHED_EPOCHS = 4;
+
+const MAX_CACHED_DISTINCT_TARGETS = 4;
+
+const INTERVALS_LATE_ATTESTATION_SUBMISSION = 1.5;
+
+const RETAIN_REGISTERED_VALIDATORS_MS = 12 * 3600 * 1000; // 12 hours
```

Comment:
Taking advantage of `onceEveryEndOfEpoch` this PR also cleans validators not seen in a while. 12 hours is an arbitrary value that seems long enough, but please feel free to suggest others